make check fails on OS X due to make build-libc

This fails on OS X, the solution is to do `brew install snaipe/soft/criterion`

* The README should be updated to include this instruction for OS X machines wishing to build libskycoin
* libskycoin should not be built by default in `make check`.  Travis should perform the libskycoin test separately

```
mkdir -p build/usr/tmp build/usr/lib build/usr/include
mkdir -p build/libskycoin bin include
rm -Rf build/libskycoin/*
go build -buildmode=c-shared  -o build/libskycoin/libskycoin.so ./lib/cgo/main.go
go build -buildmode=c-archive -o build/libskycoin/libskycoin.a  ./lib/cgo/main.go
mv build/libskycoin/libskycoin.h include/
cp lib/cgo/tests/*.c build/libskycoin/
gcc -o bin/test_libskycoin_shared build/libskycoin/*.c -lskycoin                    -lcriterion -Iinclude -Ibuild/usr/include -L build/libskycoin -Lbuild/usr/lib
build/libskycoin/check_cipher.c:2:10: fatal error: 'criterion/criterion.h' file not found
#include <criterion/criterion.h>
         ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make: *** [test-libc] Error 1
```

--- a/README.md
+++ b/README.md
@@ -148,7 +148,7 @@ We have two branches: `master` and `develop`.
 Skycoin implements client libraries which export core functionality for usage from
 other programming languages. Read the corresponding README file for further details.
 
-* `lib/cgo/` - libskycoin C client library
+* `lib/cgo/` - libskycoin C client library ( [read more](lib/cgo/README.md) )
 
 ### Running Tests
 

--- a/README.md
+++ b/README.md
@@ -37,6 +37,7 @@ Skycoin is a small part of OP Redecentralize and OP Darknet Plan.
 - [Contributing a node to the network](#contributing-a-node-to-the-network)
 - [Development](#development)
     - [Modules](#modules)
+    - [Client libraries](#client-libraries)
     - [Running Tests](#running-tests)
     - [Formatting](#formatting)
     - [Code Linting](#code-linting)

--- a/lib/cgo/README.md
+++ b/lib/cgo/README.md
@@ -1,0 +1,40 @@
+
+# Skycoin C client library
+
+[![Build Status](https://travis-ci.org/skycoin/skycoin.svg)](https://travis-ci.org/skycoin/skycoin)
+[![GoDoc](https://godoc.org/github.com/skycoin/skycoin?status.svg)](https://godoc.org/github.com/skycoin/skycoin)
+[![Go Report Card](https://goreportcard.com/badge/github.com/skycoin/skycoin)](https://goreportcard.com/report/github.com/skycoin/skycoin)
+
+Skycoin C client library (a.k.a libskycoin) provides access to Skicoin Core
+internal and API functions for implementing third-party applications.
+
+## Building
+
+```sh
+
+$ make build-libc
+```
+
+## Testing
+
+In order to test the C client libraries follow these steps
+
+- Install [Criterion](https://github.com/Snaipe/Criterion)
+  * locally by executing `make instal-deps-libc` command
+  * or by [installing Criterion system-wide](https://github.com/Snaipe/Criterion#packages)
+- Run `make test-libc` command
+
+## Binary distribution
+
+The following files will be generated
+
+- `build/libskycoin.h` - Header file for including libskycoin symbols in your app code
+- `build/libskycoin.a` - Static library.
+- `build/libskycoin.so` - Shared library object.
+
+In Mac OS X the linker will need extra `-framework CoreFoundation -framework Security`
+options.
+
+In GNU/Linux distributions it will be necessary to load symbols in `pthread`
+library e.g. by supplying extra `-lpthread` to the linker toolchain.
+

